lib/lib_opml/src/LibOpml/LibOpml.php

```diff
@@ -0,0 +1,682 @@
+<?php
+
+namespace marienfressinaud\LibOpml;
+
+/**
+ * The LibOpml class provides the methods to read and write OPML files and
+ * strings. It transforms OPML files or strings to PHP arrays (or the reverse).
+ *
+ * How to read this file?
+ *
+ * The first methods are dedicated to the parsing, and the next ones to the
+ * reading. The two last methods are helpful methods, but you don't have to
+ * worry too much about them.
+ *
+ * The main methods are the public ones: parseFile, parseString and render.
+ * They call the other parse* and render* methods internally.
+ *
+ * These three main methods are available as functions (see the src/functions.php
+ * file).
+ *
+ * What's the array format?
+ *
+ * As said before, LibOpml transforms OPML to PHP arrays, or the reverse. The
+ * format is pretty simple. It contains four keys:
+ *
+ * - version: the version of the OPML;
+ * - namespaces: an array of namespaces used in the OPML, if any;
+ * - head: an array of OPML head elements, where keys are the names of the
+ *   elements;
+ * - body: an array of arrays representing OPML outlines, where keys are the
+ *   name of the attributes (the special @outlines key contains the sub-outlines).
+ *
+ * When rendering, only the body key is required (version will default to 2.0).
+ *
+ * Example:
+ *
+ * [
+ *     version => '2.0',
+ *     namespaces => [],
+ *     head => [
+ *         title => 'An OPML file'
+ *     ],
+ *     body => [
+ *         [
+ *             text => 'Newspapers',
+ *             @outlines => [
+ *                 [text => 'El País'],
+ *                 [text => 'Le Monde'],
+ *                 [text => 'The Guardian'],
+ *                 [text => 'The New York Times'],
+ *             ]
+ *         ]
+ *     ]
+ * ]
+ *
+ * @see http://opml.org/spec2.opml
+ *
+ * @author Marien Fressinaud <dev@marienfressinaud.fr>
+ * @link https://framagit.org/marienfressinaud/lib_opml
+ * @license MIT
+ */
+class LibOpml
+{
+    /**
+     * The list of valid head elements.
+     */
+    public const HEAD_ELEMENTS = [
```

Comment:
With PHP 7.0:
```
PHP Parse error:  syntax error, unexpected 'const' (T_CONST), expecting variable (T_VARIABLE) in ./lib/marienfressinaud/lib_opml/src/LibOpml/LibOpml.php on line 67
```

I am not sure of the added value of `public` here but if it stays, this PR will have to wait till we drop PHP 7.0, which is soon: https://github.com/FreshRSS/FreshRSS/milestones

1.20.x will be the last supporting PHP 7.0 with PHP 7.2+ required from 1.21.0

Reply:
The linter asks me to add visibility to the constants. I'm a bit reluctant to change the default configuration so I guess we'll wait for the 1.21 if you don't mind :sweat_smile: I commit myself to update the PR if there are conflicts with the future changes :)

Reply:
Btw, I realize I officially only support PHP 7.4+ with lib\_opml, I'll see what I can do to support 7.2+

Reply:
PHP 7.2+ is now officially supported by lib\_opml: https://framagit.org/marienfressinaud/lib_opml/-/commit/a8dfa46a80202f3b5e71555e05eb7a4125d74b5b

Reply:
Ready for PHP 7.2+ https://github.com/FreshRSS/FreshRSS/pull/4848